iothub/device/tests/Mqtt/MqttIotHubAdapterTest.cs

```diff
@@ -173,5 +175,70 @@ private async Task SendRequestAndAcknowledgementsInSpecificOrder<T>(T requestPac
             // Assert: No matter the event ordering, sending should be awaitable without errors
             await sendRequest.ConfigureAwait(false);
         }
+
+        [TestMethod]
+        public void SettingPnpModelIdShouldSetPreviewApiVersionAndModelIdOnConnectPacket()
```

Comment:
Should probably add one more test for other protocols with setting ClientOptions and making sure it works as expected.